src/components/Button/Button.test.tsx

```diff
@@ -1,101 +1,121 @@
+    const onClick = jest.fn();
+    render(<Button element="a" disabled={true} href="" onClick={onClick} />);
+    const button = screen.getByRole("link");
+    const clickEvent = createEvent.click(button);
+    fireEvent(button, clickEvent);
+    expect(clickEvent.defaultPrevented).toBe(true);
+  });
+
   it("correctly handle icons", () => {
-    const wrapper = shallow(<Button hasIcon={true} />);
-    expect(wrapper.prop("className").includes("has-icon")).toBe(true);
+    render(<Button hasIcon={true} />);
+    expect(screen.getByRole("button")).toHaveClass("has-icon");
   });
 
   it("can be inline", () => {
-    const wrapper = shallow(<Button inline={true} />);
-    expect(wrapper.prop("className").includes("is-inline")).toBe(true);
+    render(<Button inline={true} />);
+    expect(screen.getByRole("button")).toHaveClass("is-inline");
   });
 
   it("can be dense", () => {
-    const wrapper = shallow(<Button dense />);
-    expect(wrapper.prop("className").includes("is-dense")).toBe(true);
+    render(<Button dense />);
+    expect(screen.getByRole("button")).toHaveClass("is-dense");
   });
 
   it("can be small", () => {
-    const wrapper = shallow(<Button small />);
-    expect(wrapper.prop("className").includes("is-small")).toBe(true);
+    render(<Button small />);
+    expect(screen.getByRole("button")).toHaveClass("is-small");
   });
 
   it("can add additional classes", () => {
-    const wrapper = shallow(<Button className="extra-class" />);
-    const className = wrapper.prop("className");
-    expect(className.includes("p-button")).toBe(true);
-    expect(className.includes("extra-class")).toBe(true);
+    render(<Button className="extra-class" />);
+    const button = screen.getByRole("button");
+    expect(button).toHaveClass("p-button");
+    expect(button).toHaveClass("extra-class");
   });
 
   it("puts additional classes at the end", () => {
-    const wrapper = shallow(<Button className="extra-class" dense />);
-    expect(wrapper.prop("className")).toEqual("p-button is-dense extra-class");
+    render(<Button className="extra-class" dense />);
+    expect(screen.getByRole("button").className).toEqual(
+      "p-button is-dense extra-class"
+    );
   });
 
-  it("handles base button props", () => {
-    const wrapper = shallow(<Button type="button" />);
-    wrapper.simulate("click");
-    expect(wrapper.prop("type")).toBe("button");
+  it("handles base button props", async () => {
+    render(<Button type="button" />);
+    expect(screen.getByRole("button")).toBeInTheDocument();
   });
 
-  it("handles alternate element props", () => {
-    const wrapper = shallow(<Button element="a" href="http://example.com" />);
-    wrapper.simulate("click");
-    expect(wrapper.prop("href")).toBe("http://example.com");
+  it("handles alternate element props", async () => {
+    render(<Button element="a" href="http://example.com" />);
+    const button = screen.getByRole("link");
+    expect(button).toBeInTheDocument();
```

Comment:
Haven't we just queried the button from the document? Do we need to assert if it is in document?